src/store/modules/product/actions.ts

```diff
@@ -46,48 +47,64 @@ const actions: ActionTree<ProductState, RootState> = {
     return resp;
   },
 
-  async fetchProductFacets({ commit }) {
-    const payload = {
-      "json": {
-        "query": "*:*",
-        "filter": "docType: PRODUCT",
-        "facet": {
-          "tagsFacet": {
-            "field": "tags",
-            "mincount": 1,
-            "limit": -1,
-            "sort": "index",
-            "type": "terms"
-          },
-          "productStoreIdFacet": {
-            "field": "productStoreIds",
-            "limit": -1,
-            "sort": "index",
-            "type": "terms"
-          },
-          "productCategoryNameFacet": {
-            "field": "productCategoryNames",
-            "limit": -1,
-            "sort": "index",
-            "type": "terms"
-          }
-        }
-      }
+  async updateAppliedFilters({ commit, state, dispatch }, payload) {
+    const value = payload.value
+    const appliedFilters = JSON.parse(JSON.stringify((state.appliedFilters as any)[payload.type][payload.id]))
+    appliedFilters.includes(value) ? appliedFilters.splice(appliedFilters.indexOf(value), 1) : appliedFilters.push(value)
+    commit(types.PRODUCT_APPLIED_FILTERS_UPDATED, {id: payload.id, type: payload.type, value: appliedFilters})
+    dispatch('updateQuery')
+  },
+
+  async updateQuery({ commit, dispatch, state }, payload) {
```

Comment:
We could implement a method to prepare query and have only 2 actions update filter & get products. Could be a separate task